### PR TITLE
Update to eth-utils 1.6.4 and fix new type warnings

### PR DESCRIPTION
**Pull request by palango**

### CODE REVIEW

**Comment by karlb** [comment]
Won't this cause conflicts with raiden, which uses an older eth-utils?

**Comment by palango** [comment]
Ah, I just wanted to proudly present the PR 🤷‍♂

File: src/pathfinding_service/api.py [hunk 1]
```diff
@@ -362,10 +367,10 @@ def get(self) -> Tuple[dict, int]:
             "payment_address": to_checksum_address(self.pathfinding_service.address),
         }
         if info["message"] == DEFAULT_INFO_MESSAGE:
-            info["message"] = info["message"] + to_checksum_address(
+            info["message"] = info["message"] + to_checksum_address(  # type: ignore
```
Comment (karlb) [inline src/pathfinding_service/api.py]: Why do we need the type ignore?

File: src/pathfinding_service/database.py [hunk 2]
```diff
@@ -216,7 +218,9 @@ def get_feedback_token(
 
         if token:
             return FeedbackToken(
-                token_network_address=decode_hex(token["token_network_address"]),
+                token_network_address=TokenNetworkAddress(
+                    decode_hex(token["token_network_address"])
```
Comment (karlb) [inline src/pathfinding_service/database.py]: Above we use `TokenNetworkAddress(to_canonical_address(...))`. Using the same would be more consistent.

File: src/raiden_libs/blockchain.py [hunk 3]
```diff
@@ -147,14 +148,17 @@ def get_blockchain_events(
 
     events: List[Event] = []
     for event_dict in registry_events:
+        token_network_address = TokenNetworkAddress(
+            decode_hex(event_dict["args"]["token_network_address"])
```
Comment (karlb) [inline src/raiden_libs/blockchain.py]: Again, `to_canonical_address` would be more consistent then using `decode_hex` which is just possible due to the typecast.

File: src/raiden_libs/blockchain.py [hunk 4]
```diff
@@ -218,19 +222,21 @@ def get_monitoring_blockchain_events(
         if event_name == MonitoringServiceEvent.NEW_BALANCE_PROOF_RECEIVED:
             events.append(
                 ReceiveMonitoringNewBalanceProofEvent(
-                    token_network_address=decode_hex(event["args"]["token_network_address"]),
+                    token_network_address=TokenNetworkAddress(
+                        decode_hex(event["args"]["token_network_address"])
```
Comment (karlb) [inline src/raiden_libs/blockchain.py]: Again, to_canonical_address would be more consistent then using decode_hex which is just possible due to the typecast.

File: src/raiden_libs/matrix.py [hunk 5]
```diff
@@ -284,6 +284,9 @@ def _handle_message(self, room: Any, event: dict) -> bool:
             return False
 
         for message in messages:
+            if message.sender is None:
```
Comment (karlb) [inline src/raiden_libs/matrix.py]: Can that happen? If not, I would prefer an assert.

File: src/pathfinding_service/api.py (hunk 1 above)
Comment (palango) [reply src/pathfinding_service/api.py]: Without that I get:
```
src/pathfinding_service/api.py:370: error: Unsupported left operand type for + ("object")
```

File: src/pathfinding_service/database.py (hunk 2 above)
Comment (palango) [reply src/pathfinding_service/database.py]: True, I missed that during my automatic replace.

File: src/raiden_libs/matrix.py (hunk 5 above)
Comment (palango) [reply src/raiden_libs/matrix.py]: Just checked, it cannot happen. This is checked in `raiden_libs.matrix.deserialize_messages`

File: src/pathfinding_service/api.py (hunk 1 above)
Comment (karlb) [reply src/pathfinding_service/api.py]: I would prefer
```py
info["message"] = str(info["message"]) + ...
```
or
```py
assert isinstance(info['message'], str)
```
if mypy understands that. Type ignores are always a bit distracting.

Comment (palango) [reply src/pathfinding_service/api.py]: 👍

**Pull request merged**